From 02a1c9ed5fc6628da4965c6156d8a8b5ba310d88 Mon Sep 17 00:00:00 2001
From: Supreeth Venkatesh <supreeth.venkatesh@amd.com>
Date: Thu, 10 Nov 2022 11:43:39 -0600
Subject: [PATCH 17/19] ARM: dts: aspeed: Initial device tree for AMD Siena
 Platforms
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 7bit

Initial introduction of AMD Siena platforms equipped with an
DC-SCM card which contains Aspeed ast2600 BMC manufactured by AMD.

AMD Siena platform is an AMD customer reference board with an
Aspeed ast2600 BMC manufactured by AMD.

This adds AMD Cinnabar, Shale and Sunstone device tree files
including the flash layout used by AMD Siena BMC machines.

Signed-off-by: Mohsen Dolaty <mohsen.dolaty@amd.com>
Signed-off-by: Supreeth Venkatesh <supreeth.venkatesh@amd.com>
---
 arch/arm/boot/dts/Makefile                    |   3 +
 arch/arm/boot/dts/aspeed-bmc-amd-cinnabar.dts | 827 ++++++++++++++++
 arch/arm/boot/dts/aspeed-bmc-amd-shale.dts    | 814 +++++++++++++++
 arch/arm/boot/dts/aspeed-bmc-amd-sunstone.dts | 923 ++++++++++++++++++
 4 files changed, 2567 insertions(+)
 create mode 100644 arch/arm/boot/dts/aspeed-bmc-amd-cinnabar.dts
 create mode 100644 arch/arm/boot/dts/aspeed-bmc-amd-shale.dts
 create mode 100644 arch/arm/boot/dts/aspeed-bmc-amd-sunstone.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 6613a9b1d8e4..77e9107a9bc1 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -1389,6 +1389,9 @@ dtb-$(CONFIG_ARCH_ASPEED) += \
 	aspeed-bmc-amd-quartz.dtb \
 	aspeed-bmc-amd-ruby.dtb \
 	aspeed-bmc-amd-titanite.dtb \
+	aspeed-bmc-amd-cinnabar.dtb \
+	aspeed-bmc-amd-shale.dtb \
+	aspeed-bmc-amd-sunstone.dtb \
 	aspeed-bmc-ampere-mtjade.dtb \
 	aspeed-bmc-arm-centriq2400-rep.dtb \
 	aspeed-bmc-arm-stardragon4800-rep2.dtb \
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-cinnabar.dts b/arch/arm/boot/dts/aspeed-bmc-amd-cinnabar.dts
new file mode 100644
index 000000000000..bce94407c975
--- /dev/null
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-cinnabar.dts
@@ -0,0 +1,827 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2022 AMD Inc.
+// Author: Ashok Kumar <ashok2.kumar@amd.com>
+// Author: Supreeth Venkatesh <Supreeth.Venkatesh@amd.com>
+
+/dts-v1/;
+
+#include "aspeed-g6.dtsi"
+#include "dt-bindings/gpio/aspeed-gpio.h"
+
+/ {
+	model = "AMD Cinnabar BMC";
+	compatible = "amd,cinnabar-bmc", "aspeed,ast2600";
+
+	aliases {
+		serial4 = &uart5;
+		ethernet0 = &mac3;
+		ethernet1 = &mac2;
+	};
+
+	chosen {
+		stdout-path = &uart5;
+		bootargs = "console=ttyS4,115200 earlycon vmalloc=512MB";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		fault {
+			gpios = <&gpio0 ASPEED_GPIO(O, 6) GPIO_ACTIVE_HIGH>;
+		};
+
+		identify {
+			gpios = <&gpio0 ASPEED_GPIO(O, 7) GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	memory@80000000 {
+		device_type = "memory";
+		reg = <0x80000000 0x80000000>;
+	};
+
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		video_engine_memory: jpegbuffer {
+			size = <0x02000000>;	/* 32M */
+			alignment = <0x01000000>;
+			compatible = "shared-dma-pool";
+			reusable;
+		};
+
+		pcc_memory: pccbuffer {
+			no-map;
+			reg = <0xE0000000 0x00001000>; /* 4K */
+		};
+	};
+
+	iio-hwmon-adc121c {
+		compatible = "iio-hwmon";
+		oemname0 = "ADC121C_50";
+		oemname1 = "ADC121C_51";
+		oemname2 = "ADC121C_52";
+		oemname3 = "ADC121C_53";
+		oemname4 = "ADC121C_54";
+		oemname5 = "ADC121C_55";
+		io-channels = <&adc0 0>, <&adc1 0>, <&adc2 0>,
+				<&adc3 0>, <&adc4 0>, <&adc5 0>;
+	};
+};
+
+&mctp0 {
+	status	= "okay";
+};
+
+&mdio0 {
+	status = "okay";
+
+	ethphy0: ethernet-phy@0 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <0>;
+	};
+};
+
+&mac2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rmii3_default>;
+	clock-names = "MACCLK", "RCLK";
+	use-ncsi;
+};
+
+&mac3 {
+	status = "okay";
+	phy-mode = "rgmii";
+	phy-handle = <&ethphy0>;
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rgmii4_default>;
+};
+
+&fmc {
+	status = "okay";
+	fmc-spi-user-mode;
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		status = "okay";
+		label = "bmc";
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+		#include "openbmc-flash-layout-128.dtsi"
+	};
+};
+
+&spi1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi1_default &pinctrl_qspi1_default>;
+	fmc-spi-user-mode;
+	compatible = "aspeed,ast2600-spi";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		label = "pnor";
+		status = "okay";
+	};
+
+};
+
+&spi2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi2_default &pinctrl_qspi2_default>;
+	fmc-spi-user-mode;
+	compatible = "aspeed,ast2600-spi";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		label = "pnor";
+		status = "okay";
+	};
+
+};
+
+// Host Serial console
+&uart1 {
+	compatible = "aspeed,ast2600-uart";
+	/delete-property/ reg-shift;
+	/delete-property/ reg-io-width;
+	dma-mode;
+	dma-channel = <0>;
+	status = "okay";
+};
+
+// BMC Serial Console
+&uart5 {
+	status = "okay";
+};
+
+// I2C configs
+&i2c0 {
+	//Net name i2c9 (SCM_I2C9)- MB FPGA slave
+	status = "okay";
+};
+
+&i2c1 {
+	//Net name i2c12 (SCM_I2C12) - LCD at 50KHz
+	status = "okay";
+#ifdef OPENBMC_LCD_DRIVER
+	i2cswitch@70 {
+		compatible = "nxp,pca9546";
+		reg = <0x70>;
+		//<TBD> Define LCD details here
+	};
+#endif //OPENBMC_LCD_DRIVER
+};
+
+// Misc Regulators
+&i2c4 {
+	// Net name i2c1
+	status = "okay";
+	vdd33run@13 {
+		//VDD_33_RUN VRM
+		compatible = "infineon,xdpe12284";
+		reg = <0x13>;
+	};
+	vdd33dual@14 {
+		//VDD_33_DUAL VRM
+		compatible = "infineon,xdpe12284";
+		reg = <0x14>;
+	};
+	vdd5dual@15 {
+		//VDD_5_DUAL VRM
+		compatible = "infineon,xdpe12284";
+		reg = <0x15>;
+	};
+};
+
+// P0 VRs
+&i2c5 {
+	// Net name i2c2 (SCM_I2C2)
+	status = "okay";
+	i2cswitch@71 {
+		compatible = "nxp,pca9546";
+		reg = <0x71>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		i2c@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+
+			p0_vdd_soc_run@61 {
+				//VDD SOC RUN
+				compatible = "isil,isl68137";
+				reg = <0x61>;
+			};
+
+			p0_vdd_11_sus@62 {
+				//VDD 11 SUS
+				compatible = "isil,isl68137";
+				reg = <0x62>;
+			};
+
+			p0_vdd_vddio_run@63 {
+				//VDD VDDIO RUN
+				compatible = "isil,isl68137";
+				reg = <0x63>;
+			};
+
+			p0_vdd_18_dual@64 {
+				//VDD 18 DUAL
+				compatible = "isil,isl68137";
+				reg = <0x64>;
+			};
+
+			p0_vdd_33_dual@65 {
+				//VDD 33 DUAL VRM
+				compatible = "isil,isl68137";
+				reg = <0x65>;
+			};
+		};
+	};
+};
+
+// Unused
+&i2c6 {
+	// Net name i2c3 (SCM_I2C3)
+	status = "okay";
+};
+
+// HPM Board ID
+&i2c7 {
+	// Net name i2c4 (SCM_I2C4)
+	status = "okay";
+
+	mbeeprom@50 {
+		compatible = "microchip,24lc256","atmel,24c256";
+		reg = <0x50>;
+	};
+};
+
+// PSUs
+&i2c8 {
+	// Net name i2c5 (SCM_I2C5)
+	status = "okay";
+	psueeprom@57 {
+		compatible = "microchip,24lc256","atmel,24c256";
+		reg = <0x57>;
+	};
+};
+
+// Fan, Temp, ADC sensors
+&i2c9 {
+	// Net name i2c6 (SCM_I2C6)
+	status = "okay";
+
+	i2cswitch@70 {
+		compatible = "nxp,pca9548";
+		reg = <0x70>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		i2c@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305_1_ap@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+			};
+		};
+		i2c@3 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <3>;
+
+			lm75a@48 {	// FAN FP
+				compatible = "national,lm75a";
+				reg = <0x48>;
+			};
+		};
+
+		i2c@5 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <5>;
+
+			lm75a@48 {	// PWR_CONN
+				compatible = "national,lm75a";
+				reg = <0x48>;
+			};
+
+			lm75a@49 {	// SOUTH VR
+				compatible = "national,lm75a";
+				reg = <0x49>;
+			};
+
+			lm75a@4a {	// WEST DIMM
+				compatible = "national,lm75a";
+				reg = <0x4a>;
+			};
+
+			lm75a@4b {	// EAST M.2
+				compatible = "national,lm75a";
+				reg = <0x4b>;
+			};
+
+			lm75a@4c {	// VDD_RUN
+				compatible = "national,lm75a";
+				reg = <0x4c>;
+			};
+
+			lm75a@4d {	// WEST M.2
+				compatible = "national,lm75a";
+				reg = <0x4d>;
+			};
+
+			lm75a@4e {	// VDD_DUAL
+				compatible = "national,lm75a";
+				reg = <0x4e>;
+			};
+
+			lm75a@4f {	// CHASSIS FAN
+				compatible = "national,lm75a";
+				reg = <0x4f>;
+			};
+		};
+
+		i2c@6 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <6>;
+
+			adc0: adc121c_50@50 {
+				compatible = "ti,adc121c";
+				reg = <0x50>;
+				#io-channel-cells = <1>;
+				label = "adc121c_50";
+			};
+			adc1: adc121c_51@51 {
+				compatible = "ti,adc121c";
+				reg = <0x51>;
+				#io-channel-cells = <1>;
+				label = "adc121c_51";
+			};
+			adc2: adc121c_52@52 {
+				compatible = "ti,adc121c";
+				reg = <0x52>;
+				#io-channel-cells = <1>;
+				label = "adc121c_52";
+			};
+			adc3: adc121c_53@53 {
+				compatible = "ti,adc121c";
+				reg = <0x53>;
+				#io-channel-cells = <1>;
+				label = "adc121c_53@53";
+			};
+			adc4: adc121c_54@54 {
+				compatible = "ti,adc121c";
+				reg = <0x54>;
+				#io-channel-cells = <1>;
+				label = "adc121c_54";
+			};
+			adc5: adc121c_55@55 {
+				compatible = "ti,adc121c";
+				reg = <0x55>;
+				#io-channel-cells = <1>;
+				label = "adc121c_55";
+			};
+		};
+		i2c@7 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <7>;
+			// SOC Thermal Diode
+		};
+	};
+};
+
+&i2c10 {
+	// Net name i2c7 (SCM_I2C7)
+	status = "okay";
+
+#ifdef EEPROM_PROG_ENABLE
+	i2cswitch@70 {
+		compatible = "nxp,pca9548";
+		reg = <0x70>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+	};
+
+	i2cswitch@71 {
+		compatible = "nxp,pca9546";
+		reg = <0x71>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		i2c@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+
+			brdreveeprom@50 {
+				//P0 Cinnabar BRD ID
+				compatible = "microchip,24lc256","atmel,24c256";
+				reg = <0x50>;
+			};
+
+			scmbrdeeprom@51 {
+				//P0 SCM BRD ID
+				compatible = "microchip,24lc256","atmel,24c256";
+				reg = <0x51>;
+			};
+		};
+
+	i2cswitch@72 {
+		compatible = "nxp,pca9546";
+		reg = <0x72>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		// PCIe Slots 1, 3, 4
+	};
+#endif
+};
+
+&i2c11 {
+	// Net name i2c8 (SCM_I2C8) - MB LOM
+	status = "okay";
+};
+
+&i2c12 {
+	// LOCAL MGMT - FPGA slave
+	status = "okay";
+};
+
+&i2c14 {
+	// LOCAL MGMT - EEPROM, RTC
+	status = "okay";
+	bmceeprom@50 {
+	compatible = "microchip,24lc512","atmel,24c512";
+	reg = <0x50>;
+	};
+};
+
+&i2c15 {
+	// RoT
+	status = "okay";
+};
+
+// i3c
+
+&i3c0 {
+	// P0 DIMM A0, B0, D0 SPD Access
+	status = "okay";
+
+	jdec-spd;
+	bus_id = <0x00>;
+
+	//DIMM A0
+	spd5118_0_0: spd@50,3C000000000 {
+		reg = <0x50 0x3C0 0x00000000>;
+		assigned-address = <0x50>;
+	};
+	pmic5xxx_0_0: pmic@48,20400000000 {
+		reg = <0x48 0x204 0x00000000>;
+		assigned-address = <0x48>;
+	};
+	rcd_0_0: rcd@58,2C000000000 {
+		reg = <0x58 0x2C0 0x00000000>;
+		assigned-address = <0x58>;
+	};
+	//DIMM B0
+	spd5118_0_1: spd@51,3C000000001 {
+		reg = <0x51 0x3C0 0x00000001>;
+		assigned-address = <0x51>;
+	};
+	pmic5xxx_0_1: pmic@49,20400000001 {
+		reg = <0x49 0x204 0x00000001>;
+		assigned-address = <0x49>;
+	};
+	rcd_0_1: rcd@59,2C000000001 {
+		reg = <0x59 0x2C0 0x00000001>;
+		assigned-address = <0x59>;
+	};
+	//DIMM D0
+	spd5118_0_2: spd@52,3C000000002 {
+		reg = <0x52 0x3C0 0x00000002>;
+		assigned-address = <0x52>;
+	};
+	pmic5xxx_0_2: pmic@4a,20400000002 {
+		reg = <0x4a 0x204 0x00000002>;
+		assigned-address = <0x4a>;
+	};
+	rcd_0_2: rcd@5a,2C000000002 {
+		reg = <0x5a 0x2C0 0x00000002>;
+		assigned-address = <0x5a>;
+	};
+};
+
+&i3c1 {
+	// P0 DIMM E0, F0, H0 SPD Access
+	status = "okay";
+
+	jdec-spd;
+	bus_id = <0x01>;
+
+	//DIMM E0
+	spd5118_1_0: spd@50,3C000000000 {
+		reg = <0x50 0x3C0 0x00000000>;
+		assigned-address = <0x50>;
+	};
+	pmic5xxx_1_0: pmic@48,20400000000 {
+		reg = <0x48 0x204 0x00000000>;
+		assigned-address = <0x48>;
+	};
+	rcd_1_0: rcd@58,2C000000000 {
+		reg = <0x58 0x2C0 0x00000000>;
+		assigned-address = <0x58>;
+	};
+	//DIMM F0
+	spd5118_1_1: spd@51,3C000000001 {
+		reg = <0x51 0x3C0 0x00000001>;
+		assigned-address = <0x51>;
+	};
+	pmic5xxx_1_1: pmic@49,20400000001 {
+		reg = <0x49 0x204 0x00000001>;
+		assigned-address = <0x49>;
+	};
+	rcd_1_1: rcd@59,2C000000001 {
+		reg = <0x59 0x2C0 0x00000001>;
+		assigned-address = <0x59>;
+	};
+	//DIMM H0
+	spd5118_1_2: spd@52,3C000000002 {
+		reg = <0x52 0x3C0 0x00000002>;
+		assigned-address = <0x52>;
+	};
+	pmic5xxx_1_2: pmic@4a,20400000002 {
+		reg = <0x4a 0x204 0x00000002>;
+		assigned-address = <0x4a>;
+	};
+	rcd_1_2: rcd@5a,2C000000002 {
+		reg = <0x5a 0x2C0 0x00000002>;
+		assigned-address = <0x5a>;
+	};
+ };
+
+&i3c2 {
+	// P0 DIMM A1, B1, D1 SPD Access
+	status = "okay";
+
+	jdec-spd;
+	bus_id = <0x02>;
+
+	//DIMM A1
+	spd5118_2_0: spd@50,3C000000000 {
+		reg = <0x50 0x3C0 0x00000000>;
+		assigned-address = <0x50>;
+	};
+	pmic5xxx_2_0: pmic@48,20400000000 {
+		reg = <0x48 0x204 0x00000000>;
+		assigned-address = <0x48>;
+	};
+	rcd_2_0: rcd@58,2C000000000 {
+		reg = <0x58 0x2C0 0x00000000>;
+		assigned-address = <0x58>;
+	};
+	//DIMM B1
+	spd5118_2_1: spd@51,3C000000001 {
+		reg = <0x51 0x3C0 0x00000001>;
+		assigned-address = <0x51>;
+	};
+	pmic5xxx_2_1: pmic@49,20400000001 {
+		reg = <0x49 0x204 0x00000001>;
+		assigned-address = <0x49>;
+	};
+	rcd_2_1: rcd@59,2C000000001 {
+		reg = <0x59 0x2C0 0x00000001>;
+		assigned-address = <0x59>;
+	};
+	//DIMM D1
+	spd5118_2_2: spd@52,3C000000002 {
+		reg = <0x52 0x3C0 0x00000002>;
+		assigned-address = <0x52>;
+	};
+	pmic5xxx_2_2: pmic@4a,20400000002 {
+		reg = <0x4a 0x204 0x00000002>;
+		assigned-address = <0x4a>;
+	};
+	rcd_2_2: rcd@5a,2C000000002 {
+		reg = <0x5a 0x2C0 0x00000002>;
+		assigned-address = <0x5a>;
+	};
+};
+
+&i3c3 {
+	// P0 DIMM E1, F1, H1 SPD Access
+	status = "okay";
+
+	jdec-spd;
+	bus_id = <0x03>;
+
+	//DIMM E1
+	spd5118_3_0: spd@50,3C000000000 {
+		reg = <0x50 0x3C0 0x00000000>;
+		assigned-address = <0x50>;
+	};
+	pmic5xxx_3_0: pmic@48,20400000000 {
+		reg = <0x48 0x204 0x00000000>;
+		assigned-address = <0x48>;
+	};
+	rcd_3_0: rcd@58,2C000000000 {
+		reg = <0x58 0x2C0 0x00000000>;
+		assigned-address = <0x58>;
+	};
+	//DIMM F1
+	spd5118_3_1: spd@51,3C000000001 {
+		reg = <0x51 0x3C0 0x00000001>;
+		assigned-address = <0x51>;
+	};
+	pmic5xxx_3_1: pmic@49,20400000001 {
+		reg = <0x49 0x204 0x00000001>;
+		assigned-address = <0x49>;
+	};
+	rcd_3_1: rcd@59,2C000000001 {
+		reg = <0x59 0x2C0 0x00000001>;
+		assigned-address = <0x59>;
+	};
+	//DIMM H1
+	spd5118_3_2: spd@52,3C000000002 {
+		reg = <0x52 0x3C0 0x00000002>;
+		assigned-address = <0x52>;
+	};
+	pmic5xxx_3_2: pmic@4a,20400000002 {
+		reg = <0x4a 0x204 0x00000002>;
+		assigned-address = <0x4a>;
+	};
+	rcd_3_2: rcd@5a,2C000000002 {
+		reg = <0x5a 0x2C0 0x00000002>;
+		assigned-address = <0x5a>;
+	};
+};
+
+#ifdef ENABLE_I2C_APML
+&i2c2 {
+	// P0 APML - i2c10
+	status = "okay";
+	bus-frequency = <400000>;
+
+	sbtsi@4c {
+		compatible = "amd,sbtsi";
+		reg = <0x4c>;
+	};
+
+	sbrmi@3c {
+		compatible = "amd,sbrmi";
+		reg = <0x3c>;
+		dimm-count = <12>;
+		dimm-ids = <0x81 0x83 0x84 0x87 0x89 0x8A 0x91 0x93 0x94 0x97 0x99 0x9A>;
+	};
+};
+
+&i2c3 {
+	// NC
+	status = "okay";
+};
+
+#else
+&i3c4 {
+	// P0 APML
+	status = "okay";
+
+	jdec-spd;
+	set_dasa;
+	bus_id = <0x04>;
+
+	/* Mux Bypassed by default on SP6, enable if added later */
+#ifdef IMX3112_MUX_PRESENT
+	imx3112_p0: i3cmux@70,4CC00000000 {
+		reg = <0x70 0x4CC 0x00000000>;
+		assigned-address = <0x70>;
+	};
+#endif
+
+	sbtsi_p0_0: sbtsi@4c,22400000001 {
+		reg = <0x4c 0x224 0x00000001>;
+		assigned-address = <0x4c>;
+	};
+	sbrmi_p0_1: sbrmi@3c,22400000002 {
+		reg = <0x3c 0x224 0x00000002>;
+		assigned-address = <0x3c>;
+		dimm-count = <12>;
+		dimm-ids = <0x81 0x83 0x84 0x87 0x89 0x8A 0x91 0x93 0x94 0x97 0x99 0x9A>;
+	};
+};
+
+#endif  // ENABLE_I2C_APML
+
+&espi_ctrl {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_espi_default &pinctrl_espialt_default>;
+
+	perif,dma-mode;
+};
+
+&espi_mmbi {
+	status = "okay";
+};
+
+&gpio0 {
+	gpio-line-names =
+	/*A0-A7*/	"","","","","","","","",
+	/*B0-B7*/	"","","","","MON_POST_COMPLETE","","","",
+	/*C0-C7*/	"","","","","","","","",
+	/*D0-D7*/	"","","","","","","","",
+	/*E0-E7*/	"","","","","","","","",
+	/*F0-F7*/	"","","","","","","","",
+	/*G0-G7*/	"","","","","","","","",
+	/*H0-H7*/	"","ASSERT_WARM_RST_BTN_L","ASSERT_SOC_RST_BTN_L","","","","","",
+	/*I0-I7*/	"","","","","","","","P0_I3C_APML_ALERT_L",
+	/*J0-J7*/	"","","","","","","","",
+	/*K0-K7*/	"","","","","","","","",
+	/*L0-L7*/	"","","","","","","","",
+	/*M0-M7*/	"","","","","","","","",
+	/*N0-N7*/	"","","","","","","PSP_SOFT_FUSE_NOTIFY","ASSERT_BMC_READY",
+	/*O0-O7*/	"","","HDT_SEL","HDT_XTRIG5","HDT_XTRIG6","JTAG_TRST_N","","",
+	/*P0-P7*/	"MON_RST_BTN_L","ASSERT_RST_BTN_L","MON_PWR_BTN_L","ASSERT_PWR_BTN_L","","ASSERT_NMI_BTN_L","MON_PWR_GOOD","",
+	/*Q0-Q7*/	"","","HDT_DBREQ_L","","BIOS_SPD_MUX_CTRL_RELEASED_L","","","",
+	/*R0-R7*/	"","","","","","","","",
+	/*S0-S7*/	"","","","","P1_DIMM_AF_ERROR","P1_DIMM_GL_ERROR","P0_DIMM_AF_ERROR","P0_DIMM_GL_ERROR",
+	/*T0-T7*/	"","","","","","","","",
+	/*U0-U7*/	"","","","","P1_I3C_APML_ALERT_L","","","",
+	/*V0-V7*/	"","","","","","","","",
+	/*W0-W7*/	"","","","","","","","",
+	/*X0-X7*/	"","","","","","","","",
+	/*Y0-Y7*/	"","","","","","","","",
+	/*Z0-Z7*/	"","","","","","","","",
+	/*AA0-AA7*/	"","","","","","","","",
+	/*AB0-AB7*/	"","","","","","","","",
+	/*AC0-AC7*/	"","","","","","","","";
+};
+
+&jtag1 {
+	status = "okay";
+};
+
+&video {
+	status = "okay";
+	memory-region = <&video_engine_memory>;
+};
+
+&vhub {
+	status = "okay";
+};
+
+&lpc_ctrl {
+	status = "okay";
+};
+
+&lpc_pcc {
+	port-addr = <0x80>;
+	dma-mode;
+	memory-region = <&pcc_memory>;
+	rec-mode = <0x1>;
+	port-addr-hbits-select = <0x1>;
+	port-addr-xbits = <0x3>;
+	status = "okay";
+};
+
+&kcs3 {
+	status = "okay";
+	aspeed,lpc-io-reg = <0xCA2>;
+};
+
+// eSPI Virtual UART - 0x3F8
+&vuart1 {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-shale.dts b/arch/arm/boot/dts/aspeed-bmc-amd-shale.dts
new file mode 100644
index 000000000000..8eff60190b9e
--- /dev/null
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-shale.dts
@@ -0,0 +1,814 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2022 AMD Inc.
+// Author: Abinaya Dhandapani (abinaya.dhandapani@amd.com)
+// Author: Supreeth Venkatesh <Supreeth.Venkatesh@amd.com>
+
+/dts-v1/;
+
+#include "aspeed-g6.dtsi"
+#include "dt-bindings/gpio/aspeed-gpio.h"
+
+/ {
+	model = "AMD Shale BMC";
+	compatible = "amd,shale-bmc", "aspeed,ast2600";
+
+	aliases {
+		serial4 = &uart5;
+		ethernet0 = &mac3;
+		ethernet1 = &mac2;
+	};
+
+	chosen {
+		stdout-path = &uart5;
+		bootargs = "console=ttyS4,115200 earlycon vmalloc=512MB";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		fault {
+			gpios = <&gpio0 ASPEED_GPIO(O, 6) GPIO_ACTIVE_HIGH>;
+		};
+
+		identify {
+			gpios = <&gpio0 ASPEED_GPIO(O, 7) GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	memory@80000000 {
+		device_type = "memory";
+		reg = <0x80000000 0x80000000>;
+	};
+
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		video_engine_memory: jpegbuffer {
+			size = <0x02000000>;	/* 32M */
+			alignment = <0x01000000>;
+			compatible = "shared-dma-pool";
+			reusable;
+		};
+
+		pcc_memory: pccbuffer {
+			no-map;
+			reg = <0xE0000000 0x00001000>; /* 4K */
+		};
+	};
+
+	iio-hwmon-adc121c {
+		compatible = "iio-hwmon";
+		oemname0 = "FPGA_VDD_CORE";
+		oemname1 = "FPGA_VDD_18";
+		oemname2 = "FPGA_VDD_33";
+		oemname3 = "LOM_VDD_33";
+		oemname4 = "LOM_VDD_12";
+		oemname5 = "VDD_12_DUAL";
+		io-channels = <&adc0 0>, <&adc1 0>, <&adc2 0>, <&adc3 0>,
+					<&adc4 0>, <&adc5 0>;
+	};
+};
+
+&mctp0 {
+	status	= "okay";
+};
+
+&mdio0 {
+	status = "okay";
+
+	ethphy0: ethernet-phy@0 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <0>;
+	};
+};
+
+&mac2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rmii3_default>;
+	clock-names = "MACCLK", "RCLK";
+	use-ncsi;
+};
+
+&mac3 {
+	status = "okay";
+	phy-mode = "rgmii";
+	phy-handle = <&ethphy0>;
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rgmii4_default>;
+};
+
+&fmc {
+	status = "okay";
+	fmc-spi-user-mode;
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		status = "okay";
+		label = "bmc";
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+		#include "openbmc-flash-layout-128.dtsi"
+	};
+};
+
+&spi1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi1_default &pinctrl_qspi1_default>;
+	fmc-spi-user-mode;
+	compatible = "aspeed,ast2600-spi";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		label = "pnor";
+		status = "okay";
+	};
+
+};
+
+&spi2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi2_default &pinctrl_qspi2_default>;
+	fmc-spi-user-mode;
+	compatible = "aspeed,ast2600-spi";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		label = "pnor";
+		status = "okay";
+	};
+
+};
+
+&uart1 {
+	//Host console
+	compatible = "aspeed,ast2600-uart";
+	/delete-property/ reg-shift;
+	/delete-property/ reg-io-width;
+	dma-mode;
+	dma-channel = <0>;
+	status = "okay";
+};
+
+//BMC Console
+&uart5 {
+	status = "okay";
+};
+
+// I2C configs
+&i2c0 {
+	//Net name i2c9 (SCM_I2C9)- MB FPGA slave
+	status = "okay";
+};
+
+&i2c1 {
+	//Net name i2c12 (SCM_I2C12) - LCD at 50KHz
+	// Connected to TCA9546A and to LCD HDR
+	status = "okay";
+#ifdef OPENBMC_LCD_DRIVER
+	i2cswitch@70 {
+		compatible = "nxp,pca9546";
+		reg = <0x70>;
+		//<TBD> Define LCD details here
+	};
+#endif //OPENBMC_LCD_DRIVER
+};
+
+// Misc Regulators
+&i2c4 {
+	// Net name i2c1
+	status = "okay";
+	vdd33run@13 {
+		//VDD_33_RUN VRM
+		compatible = "infineon,xdpe12284";
+		reg = <0x13>;
+	};
+	vdd33dual@14 {
+		//VDD_33_DUAL VRM
+		compatible = "infineon,xdpe12284";
+		reg = <0x14>;
+	};
+	vdd5dual@15 {
+		//VDD_5_DUAL VRM
+		compatible = "infineon,xdpe12284";
+		reg = <0x15>;
+	};
+};
+
+&i2c5 {
+	// Net name i2c2 (SCM_I2C2)
+	status = "okay";
+	i2cswitch@71 {
+		compatible = "nxp,pca9546";
+		reg = <0x71>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		i2c@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+
+			p0_vdd_soc_run@61 {
+				//VDD SOC RUN
+				compatible = "isil,isl68137";
+				reg = <0x61>;
+			};
+
+			p0_vdd_11_sus@62 {
+				//VDD 11 SUS
+				compatible = "isil,isl68137";
+				reg = <0x62>;
+			};
+
+			p0_vdd_vddio_run@63 {
+				//VDD VDDIO RUN
+				compatible = "isil,isl68137";
+				reg = <0x63>;
+			};
+
+			p0_vdd_18_dual@64 {
+				//VDD 18 DUAL
+				compatible = "isil,isl68137";
+				reg = <0x64>;
+			};
+
+			p0_vdd_33_dual@65 {
+				//VDD 33 DUAL VRM
+				compatible = "isil,isl68137";
+				reg = <0x65>;
+			};
+		};
+	};
+};
+
+// Unused
+&i2c6 {
+	// Net name i2c3 (SCM_I2C3)
+	status = "okay";
+};
+
+&i2c7 {
+	// Net name i2c4 (SCM_I2C4) - HPM BRD ID
+	status = "okay";
+
+	mbeeprom@50 {
+		compatible = "microchip,24lc256","atmel,24c256";
+		reg = <0x50>;
+	};
+};
+
+// PSUs
+&i2c8 {
+	// Net name i2c5 (SCM_I2C5)
+	status = "okay";
+	psueeprom@57 {
+		compatible = "microchip,24lc256","atmel,24c256";
+		reg = <0x57>;
+	};
+};
+
+&i2c9 {
+	// Net name i2c6 (SCM_I2C6) - FAN/LM75/ADC
+	status = "okay";
+
+	i2cswitch@70 {
+		compatible = "nxp,pca9548";
+		reg = <0x70>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		//  port 0 to 4 connected to 5 fan controllers which controls 5 fans each
+		i2c@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@4 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+
+		i2c@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@4 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+
+		i2c@2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@4 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+
+		i2c@3 {
+			reg = <3>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+
+		i2c@5 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <5>;
+
+			lm75a@48 {
+				compatible = "national,lm75a";
+				reg = <0x48>;
+			};
+
+			lm75a@49 {
+				compatible = "national,lm75a";
+				reg = <0x49>;
+			};
+
+			lm75a@4a {
+				compatible = "national,lm75a";
+				reg = <0x4a>;
+			};
+
+			lm75a@4b {
+				compatible = "national,lm75a";
+				reg = <0x4b>;
+			};
+
+			lm75a@4c {
+				compatible = "national,lm75a";
+				reg = <0x4c>;
+			};
+
+			lm75a@4d {
+				compatible = "national,lm75a";
+				reg = <0x4d>;
+			};
+
+			lm75a@4e {
+				compatible = "national,lm75a";
+				reg = <0x4e>;
+			};
+
+			lm75a@4f {
+				compatible = "national,lm75a";
+				reg = <0x4f>;
+			};
+		};
+
+		i2c@6 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <6>;
+
+			adc0: fpga_vdd_core@50 {
+				compatible = "ti,adc121c";
+				reg = <0x50>;
+				#io-channel-cells = <1>;
+				label = "fpga_vdd_core";
+			};
+			adc1: fpga_vdd_18@51 {
+				compatible = "ti,adc121c";
+				reg = <0x51>;
+				#io-channel-cells = <1>;
+				label = "fpga_vdd_18";
+			};
+			adc2: fpga_vdd_33@52 {
+				compatible = "ti,adc121c";
+				reg = <0x52>;
+				#io-channel-cells = <1>;
+				label = "fpga_vdd_33";
+			};
+			adc3: lom_vdd_33@54 {
+				compatible = "ti,adc121c";
+				reg = <0x54>;
+				#io-channel-cells = <1>;
+				label = "lom_vdd_33";
+			};
+			adc4: lom_vdd_12@55 {
+				compatible = "ti,adc121c";
+				reg = <0x55>;
+				#io-channel-cells = <1>;
+				label = "lom_vdd_12";
+			};
+			adc5: vdd_12_dual@56 {
+				compatible = "ti,adc121c";
+				reg = <0x56>;
+				#io-channel-cells = <1>;
+				label = "vdd_12_dual";
+			};
+		};
+		i2c@7 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <7>;
+			// SOC Thermal Diode
+		};
+	};
+};
+
+&i2c10 {
+	// Net name i2c7 (SCM_I2C7)
+	status = "okay";
+
+#ifdef EEPROM_PROG_ENABLE
+	i2cswitch@70 {
+		compatible = "nxp,pca9548";
+		reg = <0x70>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+	};
+
+	i2cswitch@71 {
+		compatible = "nxp,pca9546";
+		reg = <0x71>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		i2c@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+
+			brdreveeprom@50 {
+				//P0 SHALE BRD ID
+				compatible = "microchip,24lc256","atmel,24c256";
+				reg = <0x50>;
+			};
+
+			scmbrdeeprom@51 {
+				//P0 SCM BRD ID
+				compatible = "microchip,24lc256","atmel,24c256";
+				reg = <0x51>;
+			};
+		};
+	};
+
+#endif
+};
+
+&i2c11 {
+	// Net name i2c8 (SCM_I2C8) - MB LOM
+	status = "okay";
+};
+
+&i2c12 {
+	// LOCAL MGMT - FPGA slave
+	status = "okay";
+};
+
+&i2c14 {
+	// LOCAL MGMT - EEPROM, RTC
+	status = "okay";
+	bmceeprom@50 {
+	compatible = "microchip,24lc512","atmel,24c512";
+	reg = <0x50>;
+	};
+};
+
+&i2c15 {
+	// RoT
+	status = "okay";
+};
+
+// i3c
+
+&i3c0 {
+	// P0 DIMM A0, B0, D0 SPD Access
+	status = "okay";
+
+	jdec-spd;
+	bus_id = <0x00>;
+
+	//DIMM A0
+	spd5118_0_0: spd@50,3C000000000 {
+		reg = <0x50 0x3C0 0x00000000>;
+		assigned-address = <0x50>;
+	};
+	pmic5xxx_0_0: pmic@48,20400000000 {
+		reg = <0x48 0x204 0x00000000>;
+		assigned-address = <0x48>;
+	};
+	rcd_0_0: rcd@58,2C000000000 {
+		reg = <0x58 0x2C0 0x00000000>;
+		assigned-address = <0x58>;
+	};
+	//DIMM B0
+	spd5118_0_1: spd@51,3C000000001 {
+		reg = <0x51 0x3C0 0x00000001>;
+		assigned-address = <0x51>;
+	};
+	pmic5xxx_0_1: pmic@49,20400000001 {
+		reg = <0x49 0x204 0x00000001>;
+		assigned-address = <0x49>;
+	};
+	rcd_0_1: rcd@59,2C000000001 {
+		reg = <0x59 0x2C0 0x00000001>;
+		assigned-address = <0x59>;
+	};
+	//DIMM D0
+	spd5118_0_2: spd@52,3C000000002 {
+		reg = <0x52 0x3C0 0x00000002>;
+		assigned-address = <0x52>;
+	};
+	pmic5xxx_0_2: pmic@4a,20400000002 {
+		reg = <0x4a 0x204 0x00000002>;
+		assigned-address = <0x4a>;
+	};
+	rcd_0_2: rcd@5a,2C000000002 {
+		reg = <0x5a 0x2C0 0x00000002>;
+		assigned-address = <0x5a>;
+	};
+};
+
+&i3c1 {
+	// P0 DIMM E0, F0, H0 SPD Access
+	status = "okay";
+
+	jdec-spd;
+	bus_id = <0x01>;
+
+	//DIMM E0
+	spd5118_1_0: spd@50,3C000000000 {
+		reg = <0x50 0x3C0 0x00000000>;
+		assigned-address = <0x50>;
+	};
+	pmic5xxx_1_0: pmic@48,20400000000 {
+		reg = <0x48 0x204 0x00000000>;
+		assigned-address = <0x48>;
+	};
+	rcd_1_0: rcd@58,2C000000000 {
+		reg = <0x58 0x2C0 0x00000000>;
+		assigned-address = <0x58>;
+	};
+	//DIMM F0
+	spd5118_1_1: spd@51,3C000000001 {
+		reg = <0x51 0x3C0 0x00000001>;
+		assigned-address = <0x51>;
+	};
+	pmic5xxx_1_1: pmic@49,20400000001 {
+		reg = <0x49 0x204 0x00000001>;
+		assigned-address = <0x49>;
+	};
+	rcd_1_1: rcd@59,2C000000001 {
+		reg = <0x59 0x2C0 0x00000001>;
+		assigned-address = <0x59>;
+	};
+	//DIMM H0
+	spd5118_1_2: spd@52,3C000000002 {
+		reg = <0x52 0x3C0 0x00000002>;
+		assigned-address = <0x52>;
+	};
+	pmic5xxx_1_2: pmic@4a,20400000002 {
+		reg = <0x4a 0x204 0x00000002>;
+		assigned-address = <0x4a>;
+	};
+	rcd_1_2: rcd@5a,2C000000002 {
+		reg = <0x5a 0x2C0 0x00000002>;
+		assigned-address = <0x5a>;
+	};
+};
+
+#ifdef ENABLE_I2C_APML
+&i2c2 {
+	// P0 APML - i2c10
+	status = "okay";
+	bus-frequency = <400000>;
+
+	sbtsi@4c {
+		compatible = "amd,sbtsi";
+		reg = <0x4c>;
+	};
+
+	sbrmi@3c {
+		compatible = "amd,sbrmi";
+		reg = <0x3c>;
+		dimm-count = <6>;
+		dimm-ids = <0x81 0x83 0x84 0x87 0x89 0x8A>;
+	};
+};
+
+&i2c3 {
+	// Net Name - i2c11
+	status = "okay";
+};
+
+#else
+&i3c4 {
+	// P0 APML
+	status = "okay";
+
+	jdec-spd;
+	set_dasa;
+	bus_id = <0x04>;
+
+	/* Mux Bypassed by default on SP6, enable if added later */
+#ifdef IMX3112_MUX_PRESENT
+	imx3112_p0: i3cmux@70,4CC00000000 {
+		reg = <0x70 0x4CC 0x00000000>;
+		assigned-address = <0x70>;
+	};
+#endif
+
+	sbtsi_p0_0: sbtsi@4c,22400000001 {
+		reg = <0x4c 0x224 0x00000001>;
+		assigned-address = <0x4c>;
+	};
+	sbrmi_p0_1: sbrmi@3c,22400000002 {
+		reg = <0x3c 0x224 0x00000002>;
+		assigned-address = <0x3c>;
+		dimm-count = <6>;
+		dimm-ids = <0x81 0x83 0x84 0x87 0x89 0x8A>;
+	};
+};
+#endif  // ENABLE_I2C_APML
+
+&espi_ctrl {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_espi_default &pinctrl_espialt_default>;
+
+	perif,dma-mode;
+};
+
+&espi_mmbi {
+	status = "okay";
+};
+
+&gpio0 {
+	gpio-line-names =
+	/*A0-A7*/	"","","","","","","","",
+	/*B0-B7*/	"","","","","MON_POST_COMPLETE","","","",
+	/*C0-C7*/	"","","","","","","","",
+	/*D0-D7*/	"","","","","","","","",
+	/*E0-E7*/	"","","","","","","","",
+	/*F0-F7*/	"","","","","","","","",
+	/*G0-G7*/	"","","","","","","","",
+	/*H0-H7*/	"","ASSERT_WARM_RST_BTN_L","ASSERT_SOC_RST_BTN_L","","","","","",
+	/*I0-I7*/	"","","","","","","","P0_I3C_APML_ALERT_L",
+	/*J0-J7*/	"","","","","","","","",
+	/*K0-K7*/	"","","","","","","","",
+	/*L0-L7*/	"","","","","","","","",
+	/*M0-M7*/	"","","","","","","","",
+	/*N0-N7*/	"","","","","","","PSP_SOFT_FUSE_NOTIFY","ASSERT_BMC_READY",
+	/*O0-O7*/	"","","HDT_SEL","HDT_XTRIG5","HDT_XTRIG6","JTAG_TRST_N","","",
+	/*P0-P7*/	"MON_RST_BTN_L","ASSERT_RST_BTN_L","MON_PWR_BTN_L","ASSERT_PWR_BTN_L","","ASSERT_NMI_BTN_L","MON_PWR_GOOD","",
+	/*Q0-Q7*/	"","","HDT_DBREQ_L","","BIOS_SPD_MUX_CTRL_RELEASED_L","","","",
+	/*R0-R7*/	"","","","","","","","",
+	/*S0-S7*/	"","","","","P1_DIMM_AF_ERROR","P1_DIMM_GL_ERROR","P0_DIMM_AF_ERROR","P0_DIMM_GL_ERROR",
+	/*T0-T7*/	"","","","","","","","",
+	/*U0-U7*/	"","","","","P1_I3C_APML_ALERT_L","","","",
+	/*V0-V7*/	"","","","","","","","",
+	/*W0-W7*/	"","","","","","","","",
+	/*X0-X7*/	"","","","","","","","",
+	/*Y0-Y7*/	"","","","","","","","",
+	/*Z0-Z7*/	"","","","","","","","",
+	/*AA0-AA7*/	"","","","","","","","",
+	/*AB0-AB7*/	"","","","","","","","",
+	/*AC0-AC7*/	"","","","","","","","";
+};
+
+&jtag1 {
+	status = "okay";
+};
+
+&video {
+	status = "okay";
+	memory-region = <&video_engine_memory>;
+};
+
+&vhub {
+	status = "okay";
+};
+
+&lpc_ctrl {
+	status = "okay";
+};
+
+&lpc_pcc {
+	port-addr = <0x80>;
+	dma-mode;
+	memory-region = <&pcc_memory>;
+	rec-mode = <0x1>;
+	port-addr-hbits-select = <0x1>;
+	port-addr-xbits = <0x3>;
+	status = "okay";
+};
+
+&kcs3 {
+	status = "okay";
+	aspeed,lpc-io-reg = <0xCA2>;
+};
+
+// eSPI Virtual UART - 0x3F8
+&vuart1 {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-sunstone.dts b/arch/arm/boot/dts/aspeed-bmc-amd-sunstone.dts
new file mode 100644
index 000000000000..6791e4bf4012
--- /dev/null
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-sunstone.dts
@@ -0,0 +1,923 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2022 AMD Inc.
+// Author: Rajaganesh Rathinasabapathi <Rajaganesh.Rathinasabapathi@amd.com>
+// Author: Supreeth Venkatesh <Supreeth.Venkatesh@amd.com>
+
+/dts-v1/;
+
+#include "aspeed-g6.dtsi"
+#include "dt-bindings/gpio/aspeed-gpio.h"
+
+/ {
+	model = "AMD Sunstone BMC";
+	compatible = "amd,sunstone-bmc", "aspeed,ast2600";
+
+	aliases {
+		serial4 = &uart5;
+		ethernet0 = &mac3;
+		ethernet1 = &mac2;
+	};
+
+	chosen {
+		stdout-path = &uart5;
+		bootargs = "console=ttyS4,115200 earlycon vmalloc=512MB";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		fault {
+			gpios = <&gpio0 ASPEED_GPIO(O, 6) GPIO_ACTIVE_HIGH>;
+		};
+
+		identify {
+			gpios = <&gpio0 ASPEED_GPIO(O, 7) GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	memory@80000000 {
+		device_type = "memory";
+		reg = <0x80000000 0x80000000>;
+	};
+
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+
+		video_engine_memory: jpegbuffer {
+			size = <0x02000000>;	/* 32M */
+			alignment = <0x01000000>;
+			compatible = "shared-dma-pool";
+			reusable;
+		};
+
+		pcc_memory: pccbuffer {
+			no-map;
+			reg = <0xE0000000 0x00001000>; /* 4K */
+		};
+	};
+
+	iio-hwmon-adc121c {
+		compatible = "iio-hwmon";
+		oemname0 = "FPGA_VDD_CORE";
+		oemname1 = "FPGA_VDD_18";
+		oemname2 = "FPGA_VDD_33";
+		oemname3 = "LOM_VDD_33";
+		oemname4 = "LOM_VDD_12";
+		oemname5 = "VDD_12_DUAL";
+		io-channels = <&adc0 0>, <&adc1 0>, <&adc2 0>, <&adc3 0>, <&adc4 0>, <&adc5 0>;
+	};
+};
+
+&mctp0 {
+	status	= "okay";
+};
+
+&mdio0 {
+	status = "okay";
+
+	ethphy0: ethernet-phy@0 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <0>;
+	};
+};
+
+&mac2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rmii3_default>;
+	clock-names = "MACCLK", "RCLK";
+	use-ncsi;
+};
+
+&mac3 {
+	status = "okay";
+	phy-mode = "rgmii";
+	phy-handle = <&ethphy0>;
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_rgmii4_default>;
+};
+
+&fmc {
+	status = "okay";
+	fmc-spi-user-mode;
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		status = "okay";
+		label = "bmc";
+		spi-tx-bus-width = <4>;
+		spi-rx-bus-width = <4>;
+		#include "openbmc-flash-layout-128.dtsi"
+	};
+};
+
+&spi1 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi1_default &pinctrl_qspi1_default>;
+	fmc-spi-user-mode;
+	compatible = "aspeed,ast2600-spi";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		label = "pnor";
+		status = "okay";
+	};
+
+};
+
+&spi2 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_spi2_default &pinctrl_qspi2_default>;
+	fmc-spi-user-mode;
+	compatible = "aspeed,ast2600-spi";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		label = "pnor";
+		status = "okay";
+	};
+
+};
+
+// Host Serial console
+&uart1 {
+	compatible = "aspeed,ast2600-uart";
+	/delete-property/ reg-shift;
+	/delete-property/ reg-io-width;
+	dma-mode;
+	dma-channel = <0>;
+	status = "okay";
+};
+
+// BMC Serial Console
+&uart5 {
+	status = "okay";
+};
+
+// HPM FPGA
+&i2c0 {
+	//Net name i2c9 - MB FPGA slave
+	status = "okay";
+};
+
+// Post Code LCD
+&i2c1 {
+	// Connected to TCA9546A and to LCD HDR
+	status = "okay";
+#ifdef OPENBMC_LCD_DRIVER
+	i2cswitch@70 {
+		compatible = "nxp,pca9546";
+		reg = <0x70>;
+		//<TBD> Define LCD details here
+	};
+#endif //OPENBMC_LCD_DRIVER
+};
+
+// Misc Regulators
+&i2c4 {
+	// Net name i2c1
+	status = "okay";
+	vdd33run@13 {
+		//VDD_33_RUN VRM
+		compatible = "infineon,xdpe12284";
+		reg = <0x13>;
+	};
+	vdd33dual@14 {
+		//VDD_33_DUAL VRM
+		compatible = "infineon,xdpe12284";
+		reg = <0x14>;
+	};
+	vdd5dual@15 {
+		//VDD_5_DUAL VRM
+		compatible = "infineon,xdpe12284";
+		reg = <0x15>;
+	};
+};
+
+// P0 VRs
+&i2c5 {
+	// Net name i2c2 (SCM_I2C2)
+	status = "okay";
+	i2cswitch@71 {
+		compatible = "nxp,pca9546";
+		reg = <0x71>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		i2c@0 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0>;
+
+			p0_vdd_soc_run@40 {
+				//VDD SOC RUN
+				compatible = "infineon,xdpe12284";
+				reg = <0x40>;
+			};
+
+			p0_vdd_11_sus@42 {
+				//VDD 11 SUS
+				compatible = "infineon,xdpe12284";
+				reg = <0x42>;
+			};
+
+			p0_vdd_vddio_run@41 {
+				//VDD VDDIO RUN
+				compatible = "infineon,xdpe12284";
+				reg = <0x41>;
+			};
+
+			p0_vdd_18_dual@17 {
+				//VDD 18 DUAL
+				compatible = "infineon,xdpe12284";
+				reg = <0x17>;
+			};
+
+			p0_vdd_33_dual@16 {
+				//VDD 33 DUAL VRM
+				compatible = "infineon,xdpe12284";
+				reg = <0x16>;
+			};
+		};
+	};
+};
+
+// Unused
+&i2c6 {
+	// Net name i2c3 (SCM_I2C3)
+	status = "okay";
+};
+
+// HPM Board ID
+&i2c7 {
+	// Net name i2c4
+	status = "okay";
+
+	mbeeprom@50 {
+		compatible = "microchip,24lc256","atmel,24c256";
+		reg = <0x50>;
+	};
+};
+
+// PSUs
+&i2c8 {
+	// Net name i2c5
+	status = "okay";
+	psueeprom@57 {
+		compatible = "microchip,24lc256","atmel,24c256";
+		reg = <0x57>;
+	};
+};
+
+// Fan, Temp, ADC sensors
+&i2c9 {
+	// Net name i2c6
+	status = "okay";
+
+	i2cswitch@70 {
+		compatible = "nxp,pca9548";
+		reg = <0x70>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		// Port 0 to 4 connected to 5 fan controllers which controls 5 fans each
+		// Chassis Fans 0,1,2,3,4
+		i2c@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@4 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+		// Chassis Fans 5,6,7,8,9
+		i2c@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@4 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+		// P0 Fan, P0 Water Pump Fan, P0 NS VR Fan 1 & 2, P0 GL DIMM Fan
+		i2c@2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@2 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@4 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+			};
+		};
+		// P0 SS VR Fan 0 & 1, P0 AF DIMM Fan
+		i2c@3 {
+			reg = <3>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			emc2305@4d {
+				compatible = "smsc,emc2305";
+				reg = <0x4d>;
+				#cooling-cells = <0x02>;
+
+				fan@0 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+				fan@1 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+				fan@3 {
+					min-rpm = /bits/ 16 <1000>;
+					max-rpm = /bits/ 16 <16000>;
+				};
+
+			};
+		};
+
+		i2c@5 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <5>;
+
+			lm75a@48 {	// PWR_CONN
+				compatible = "national,lm75a";
+				reg = <0x48>;
+			};
+
+			lm75a@49 {	// SOUTH VR
+				compatible = "national,lm75a";
+				reg = <0x49>;
+			};
+
+			lm75a@4a {	// WEST DIMM
+				compatible = "national,lm75a";
+				reg = <0x4a>;
+			};
+
+			lm75a@4b {	// EAST M.2
+				compatible = "national,lm75a";
+				reg = <0x4b>;
+			};
+
+			lm75a@4c {	// VDD_RUN
+				compatible = "national,lm75a";
+				reg = <0x4c>;
+			};
+
+			lm75a@4d {	// WEST M.2
+				compatible = "national,lm75a";
+				reg = <0x4d>;
+			};
+
+			lm75a@4e {	// VDD_DUAL
+				compatible = "national,lm75a";
+				reg = <0x4e>;
+			};
+
+			lm75a@4f {	// CHASSIS FAN
+				compatible = "national,lm75a";
+				reg = <0x4f>;
+			};
+		};
+
+		i2c@6 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <6>;
+
+			adc0: fpga_vdd_core@50 {
+				compatible = "ti,adc121c";
+				reg = <0x50>;
+				#io-channel-cells = <1>;
+				label = "fpga_vdd_core";
+			};
+			adc1: fpga_vdd_18@51 {
+				compatible = "ti,adc121c";
+				reg = <0x51>;
+				#io-channel-cells = <1>;
+				label = "fpga_vdd_18";
+			};
+			adc2: fpga_vdd_33@52 {
+				compatible = "ti,adc121c";
+				reg = <0x52>;
+				#io-channel-cells = <1>;
+				label = "fpga_vdd_33";
+			};
+			adc3: lom_vdd_33@54 {
+				compatible = "ti,adc121c";
+				reg = <0x54>;
+				#io-channel-cells = <1>;
+				label = "lom_vdd_33";
+			};
+			adc4: lom_vdd_12@55 {
+				compatible = "ti,adc121c";
+				reg = <0x55>;
+				#io-channel-cells = <1>;
+				label = "lom_vdd_12";
+			};
+			adc5: vdd_12_dual@56 {
+				compatible = "ti,adc121c";
+				reg = <0x56>;
+				#io-channel-cells = <1>;
+				label = "vdd_12_dual";
+			};
+		};
+		i2c@7 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <7>;
+			// SOC Thermal Diode
+		};
+	};
+};
+
+// Sunstone brd_id and PCIe slots
+// Usually disabled as host accesses this
+&i2c10 {
+	// Net name i2c7
+	status = "okay";
+#ifdef EEPROM_PROG_ENABLE
+	i2cswitch@70 {
+		compatible = "nxp,pca9548";
+		reg = <0x70>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		// UBM
+	};
+
+	i2cswitch@71 {
+		compatible = "nxp,pca9546";
+		reg = <0x71>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		i2c@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			// BRD ID eeprom
+			brdreveeprom@50 {
+				compatible = "microchip,24lc256","atmel,24c256";
+				reg = <0x50>;
+			};
+			// SCM BRD ID eeprom
+			scmbrdeeprom@51 {
+				compatible = "microchip,24lc256","atmel,24c256";
+				reg = <0x51>;
+			};
+		};
+	};
+
+	i2cswitch@72 {
+		compatible = "nxp,pca9546";
+		reg = <0x72>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		// PCIe Slots 1, 3, 4
+	};
+#endif
+};
+
+// 1G LOM
+&i2c11 {
+	// Net name i2c8
+	status = "okay";
+};
+
+// LOCAL MGMT - FPGA slave
+&i2c12 {
+	status = "okay";
+};
+
+// LOCAL MGMT - EEPROM, RTC
+&i2c14 {
+	status = "okay";
+	bmceeprom@50 {
+		compatible = "microchip,24lc512","atmel,24c512";
+		reg = <0x50>;
+	};
+};
+
+// ROT
+&i2c15 {
+	status = "okay";
+};
+
+// i3c
+
+&i3c0 {
+	// P0 DIMM A0, B0, D0 SPD Access
+	status = "okay";
+
+	jdec-spd;
+	bus_id = <0x00>;
+
+	//DIMM A0
+	spd5118_0_0: spd@50,3C000000000 {
+		reg = <0x50 0x3C0 0x00000000>;
+		assigned-address = <0x50>;
+	};
+	pmic5xxx_0_0: pmic@48,20400000000 {
+		reg = <0x48 0x204 0x00000000>;
+		assigned-address = <0x48>;
+	};
+	rcd_0_0: rcd@58,2C000000000 {
+		reg = <0x58 0x2C0 0x00000000>;
+		assigned-address = <0x58>;
+	};
+	//DIMM B0
+	spd5118_0_1: spd@51,3C000000001 {
+		reg = <0x51 0x3C0 0x00000001>;
+		assigned-address = <0x51>;
+	};
+	pmic5xxx_0_1: pmic@49,20400000001 {
+		reg = <0x49 0x204 0x00000001>;
+		assigned-address = <0x49>;
+	};
+	rcd_0_1: rcd@59,2C000000001 {
+		reg = <0x59 0x2C0 0x00000001>;
+		assigned-address = <0x59>;
+	};
+	//DIMM D0
+	spd5118_0_2: spd@52,3C000000002 {
+		reg = <0x52 0x3C0 0x00000002>;
+		assigned-address = <0x52>;
+	};
+	pmic5xxx_0_2: pmic@4a,20400000002 {
+		reg = <0x4a 0x204 0x00000002>;
+		assigned-address = <0x4a>;
+	};
+	rcd_0_2: rcd@5a,2C000000002 {
+		reg = <0x5a 0x2C0 0x00000002>;
+		assigned-address = <0x5a>;
+	};
+};
+
+&i3c1 {
+	// P0 DIMM E0, F0, H0 SPD Access
+	status = "okay";
+
+	jdec-spd;
+	bus_id = <0x01>;
+
+	//DIMM E0
+	spd5118_1_0: spd@50,3C000000000 {
+		reg = <0x50 0x3C0 0x00000000>;
+		assigned-address = <0x50>;
+	};
+	pmic5xxx_1_0: pmic@48,20400000000 {
+		reg = <0x48 0x204 0x00000000>;
+		assigned-address = <0x48>;
+	};
+	rcd_1_0: rcd@58,2C000000000 {
+		reg = <0x58 0x2C0 0x00000000>;
+		assigned-address = <0x58>;
+	};
+	//DIMM F0
+	spd5118_1_1: spd@51,3C000000001 {
+		reg = <0x51 0x3C0 0x00000001>;
+		assigned-address = <0x51>;
+	};
+	pmic5xxx_1_1: pmic@49,20400000001 {
+		reg = <0x49 0x204 0x00000001>;
+		assigned-address = <0x49>;
+	};
+	rcd_1_1: rcd@59,2C000000001 {
+		reg = <0x59 0x2C0 0x00000001>;
+		assigned-address = <0x59>;
+	};
+	//DIMM H0
+	spd5118_1_2: spd@52,3C000000002 {
+		reg = <0x52 0x3C0 0x00000002>;
+		assigned-address = <0x52>;
+	};
+	pmic5xxx_1_2: pmic@4a,20400000002 {
+		reg = <0x4a 0x204 0x00000002>;
+		assigned-address = <0x4a>;
+	};
+	rcd_1_2: rcd@5a,2C000000002 {
+		reg = <0x5a 0x2C0 0x00000002>;
+		assigned-address = <0x5a>;
+	};
+ };
+
+&i3c2 {
+	// P0 DIMM A1, B1, D1 SPD Access
+	status = "okay";
+
+	jdec-spd;
+	bus_id = <0x02>;
+
+	//DIMM A1
+	spd5118_2_0: spd@50,3C000000000 {
+		reg = <0x50 0x3C0 0x00000000>;
+		assigned-address = <0x50>;
+	};
+	pmic5xxx_2_0: pmic@48,20400000000 {
+		reg = <0x48 0x204 0x00000000>;
+		assigned-address = <0x48>;
+	};
+	rcd_2_0: rcd@58,2C000000000 {
+		reg = <0x58 0x2C0 0x00000000>;
+		assigned-address = <0x58>;
+	};
+	//DIMM B1
+	spd5118_2_1: spd@51,3C000000001 {
+		reg = <0x51 0x3C0 0x00000001>;
+		assigned-address = <0x51>;
+	};
+	pmic5xxx_2_1: pmic@49,20400000001 {
+		reg = <0x49 0x204 0x00000001>;
+		assigned-address = <0x49>;
+	};
+	rcd_2_1: rcd@59,2C000000001 {
+		reg = <0x59 0x2C0 0x00000001>;
+		assigned-address = <0x59>;
+	};
+	//DIMM D1
+	spd5118_2_2: spd@52,3C000000002 {
+		reg = <0x52 0x3C0 0x00000002>;
+		assigned-address = <0x52>;
+	};
+	pmic5xxx_2_2: pmic@4a,20400000002 {
+		reg = <0x4a 0x204 0x00000002>;
+		assigned-address = <0x4a>;
+	};
+	rcd_2_2: rcd@5a,2C000000002 {
+		reg = <0x5a 0x2C0 0x00000002>;
+		assigned-address = <0x5a>;
+	};
+};
+
+&i3c3 {
+	// P0 DIMM E1, F1, H1 SPD Access
+	status = "okay";
+
+	jdec-spd;
+	bus_id = <0x03>;
+
+	//DIMM E1
+	spd5118_3_0: spd@50,3C000000000 {
+		reg = <0x50 0x3C0 0x00000000>;
+		assigned-address = <0x50>;
+	};
+	pmic5xxx_3_0: pmic@48,20400000000 {
+		reg = <0x48 0x204 0x00000000>;
+		assigned-address = <0x48>;
+	};
+	rcd_3_0: rcd@58,2C000000000 {
+		reg = <0x58 0x2C0 0x00000000>;
+		assigned-address = <0x58>;
+	};
+	//DIMM F1
+	spd5118_3_1: spd@51,3C000000001 {
+		reg = <0x51 0x3C0 0x00000001>;
+		assigned-address = <0x51>;
+	};
+	pmic5xxx_3_1: pmic@49,20400000001 {
+		reg = <0x49 0x204 0x00000001>;
+		assigned-address = <0x49>;
+	};
+	rcd_3_1: rcd@59,2C000000001 {
+		reg = <0x59 0x2C0 0x00000001>;
+		assigned-address = <0x59>;
+	};
+	//DIMM H1
+	spd5118_3_2: spd@52,3C000000002 {
+		reg = <0x52 0x3C0 0x00000002>;
+		assigned-address = <0x52>;
+	};
+	pmic5xxx_3_2: pmic@4a,20400000002 {
+		reg = <0x4a 0x204 0x00000002>;
+		assigned-address = <0x4a>;
+	};
+	rcd_3_2: rcd@5a,2C000000002 {
+		reg = <0x5a 0x2C0 0x00000002>;
+		assigned-address = <0x5a>;
+	};
+};
+
+#ifdef ENABLE_I2C_APML
+&i2c2 {
+	// P0 APML - i2c10
+	status = "okay";
+	bus-frequency = <400000>;
+
+	sbtsi@4c {
+		compatible = "amd,sbtsi";
+		reg = <0x4c>;
+	};
+
+	sbrmi@3c {
+		compatible = "amd,sbrmi";
+		reg = <0x3c>;
+		dimm-count = <12>;
+		dimm-ids = <0x81 0x83 0x84 0x87 0x89 0x8A 0x91 0x93 0x94 0x97 0x99 0x9A>;
+	};
+};
+
+&i2c3 {
+	// Net Name - i2c11
+	status = "okay";
+};
+
+#else
+&i3c4 {
+	// P0 APML
+	status = "okay";
+
+	jdec-spd;
+	bus_id = <0x04>;
+	set_dasa;
+	/* Mux Bypassed by default on SP6, enable if added later */
+#ifdef IMX3112_MUX_PRESENT
+	imx3112_p0: i3cmux@70,4CC00000000 {
+		reg = <0x70 0x4CC 0x00000000>;
+		assigned-address = <0x70>;
+	};
+#endif
+
+	sbtsi_p0_0: sbtsi@4c,22400000001 {
+		reg = <0x4c 0x224 0x00000001>;
+		assigned-address = <0x4c>;
+	};
+	sbrmi_p0_1: sbrmi@3c,22400000002 {
+		reg = <0x3c 0x224 0x00000002>;
+		assigned-address = <0x3c>;
+		dimm-count = <12>;
+		dimm-ids = <0x81 0x83 0x84 0x87 0x89 0x8A 0x91 0x93 0x94 0x97 0x99 0x9A>;
+	};
+};
+
+#endif  // ENABLE_I2C_APML
+
+&espi_ctrl {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_espi_default &pinctrl_espialt_default>;
+
+	perif,dma-mode;
+};
+
+&espi_mmbi {
+	status = "okay";
+};
+
+&gpio0 {
+	gpio-line-names =
+	/*A0-A7*/	"","","","","","","","",
+	/*B0-B7*/	"","","","","MON_POST_COMPLETE","","","",
+	/*C0-C7*/	"","","","","","","","",
+	/*D0-D7*/	"","","","","","","","",
+	/*E0-E7*/	"","","","","","","","",
+	/*F0-F7*/	"","","","","","","","",
+	/*G0-G7*/	"","","","","","","","",
+	/*H0-H7*/	"","ASSERT_WARM_RST_BTN_L","ASSERT_SOC_RST_BTN_L","","","","","",
+	/*I0-I7*/	"","","","","","","","P0_I3C_APML_ALERT_L",
+	/*J0-J7*/	"","","","","","","","",
+	/*K0-K7*/	"","","","","","","","",
+	/*L0-L7*/	"","","","","","","","",
+	/*M0-M7*/	"","","","","","","","",
+	/*N0-N7*/	"","","","","","","PSP_SOFT_FUSE_NOTIFY","ASSERT_BMC_READY",
+	/*O0-O7*/	"","","HDT_SEL","HDT_XTRIG5","HDT_XTRIG6","JTAG_TRST_N","","",
+	/*P0-P7*/	"MON_RST_BTN_L","ASSERT_RST_BTN_L","MON_PWR_BTN_L","ASSERT_PWR_BTN_L","","ASSERT_NMI_BTN_L","MON_PWR_GOOD","",
+	/*Q0-Q7*/	"","","HDT_DBREQ_L","","BIOS_SPD_MUX_CTRL_RELEASED_L","","","",
+	/*R0-R7*/	"","","","","","","","",
+	/*S0-S7*/	"","","","","P1_DIMM_AF_ERROR","P1_DIMM_GL_ERROR","P0_DIMM_AF_ERROR","P0_DIMM_GL_ERROR",
+	/*T0-T7*/	"","","","","","","","",
+	/*U0-U7*/	"","","","","P1_I3C_APML_ALERT_L","","","",
+	/*V0-V7*/	"","","","","","","","",
+	/*W0-W7*/	"","","","","","","","",
+	/*X0-X7*/	"","","","","","","","",
+	/*Y0-Y7*/	"","","","","","","","",
+	/*Z0-Z7*/	"","","","","","","","",
+	/*AA0-AA7*/	"","","","","","","","",
+	/*AB0-AB7*/	"","","","","","","","",
+	/*AC0-AC7*/	"","","","","","","","";
+};
+
+&jtag1 {
+	status = "okay";
+};
+
+&video {
+	status = "okay";
+	memory-region = <&video_engine_memory>;
+};
+
+&vhub {
+	status = "okay";
+};
+
+&lpc_ctrl {
+	status = "okay";
+};
+
+&lpc_pcc {
+	port-addr = <0x80>;
+	dma-mode;
+	memory-region = <&pcc_memory>;
+	rec-mode = <0x1>;
+	port-addr-hbits-select = <0x1>;
+	port-addr-xbits = <0x3>;
+	status = "okay";
+};
+
+&kcs3 {
+	status = "okay";
+	aspeed,lpc-io-reg = <0xCA2>;
+};
+
+// eSPI Virtual UART - 0x3F8
+&vuart1 {
+	status = "okay";
+};
--
2.17.1

